app/models/user_event.rb

```diff
@@ -49,7 +49,8 @@ def reorderable?
     end
 
     def reorder_priorities
-      reorderable_events = associated_user_events.order_by_priority.where(["priority > ?", priority_was.to_i])
+      priority_gt_predicate = UserEvent.arel_table[:priority].gt(priority_was.to_i)
+      reorderable_events = associated_user_events.order_by_priority.where(priority_gt_predicate)
       reorderable_events.update_all("priority = priority - 1")
```

Comment:
Can we also use arel here? Please explore the possibilities.

Reply:
It will make the update query complicated. and unreadable Since we are not taking in inputs it's safe right?

Reply:
It's not about inputs, it's just preference to not use any raw sql, instead build with arel.

Reply:
we are planning to move act_as_list gem for priority change right? I think this code we can remove from there. And I couldn't find an optimum solution using arel to make SQL queries.

found these two articles.

https://stackoverflow.com/questions/49051240/increment-attribute-with-arel-table
https://jpospisil.com/2014/06/16/the-definitive-guide-to-arel-the-sql-manager-for-ruby.html